src/common/timer/interaction-timer.js

```diff
@@ -0,0 +1,43 @@
+import { Timer } from './timer'
+import { subscribeToVisibilityChange } from '../window/page-visibility'
+import { debounce } from '../util/invoke'
+import { isBrowserScope } from '../util/global-scope'
+
+export class InteractionTimer extends Timer {
+  constructor (opts, ms) {
+    super(opts, ms)
+    this.onRefresh = opts.onRefresh
+
+    if (!opts.refreshEvents) opts.refreshEvents = ['click', 'keydown', 'scroll']
+    this.abortController = opts.abortController
+
+    if (isBrowserScope && opts.ee) {
+      if (opts.ee) {
+        const debouncedRefresh = debounce(this.refresh.bind(this), 500, { leading: true })
+        opts.ee.on('fn-end', (evts) => {
+          if (opts.refreshEvents.includes(evts?.[0]?.type) || '') {
+            debouncedRefresh()
+          }
+        })
+      }
+
+      // // watch for the vis state changing.  If the page is hidden, the local inactivity timer should be paused
+      // // if the page is brought BACK to visibility and the timer hasnt "naturally" expired, refresh the timer...
+      // // this is to support the concept that other tabs could be experiencing activity.  The thought would be that
+      // // "backgrounded" tabs would pause, while "closed" tabs that "reopen" will just instantiate a new SessionEntity class if restored
+      // // which will do a "hard" check of the timestamps.
+      subscribeToVisibilityChange((state) => {
+        if (state === 'hidden') this.pause()
+        else this.refresh()
+      }, false, false, this.abortController?.signal)
```

Comment:
Timers are local to the page. We don't want them to arbitrarily clear the session if the page is open on another tab. They need to be paused to prevent cross contamination if multiple pages are open

Reply:
will add check to read on vis change "shown"